File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s == "") {
            return true;
        }

        int idx_in_s = 0;

        for (int i = 0; i < t.size(); i++) {
            if (t[i] == s[idx_in_s]) {
                idx_in_s++;

                if (idx_in_s == s.size()) {
                    break;
                }
            }
        }

        if (idx_in_s == s.size()) return true;
        return false;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.length() == 0) return true;

        if (t.length() == 0) return false;

        int sIndex = 0;
        int tIndex = 0;

        while (tIndex < t.length() && sIndex < s.length()) {
            if (s.charAt(sIndex) == t.charAt(tIndex)) {
                sIndex++;
            }

            tIndex++;
        }

        return sIndex == s.length();
    }
}